tests/test_precision.py

```diff
@@ -87,3 +93,13 @@ def test_predict_precision_memory(precision: Precision):
     memory_fp32 = predict_and_measure_memory(Precision.FP32)
     memory_half = predict_and_measure_memory(precision)
     assert memory_half < 0.95 * memory_fp32
+
+
+def test_amp_fp8_path():
+    trainer = get_trainer(Precision.AMP_FP8)
+    if te_installed:
+        with pytest.raises(RuntimeError, match='AMP_FP8 precision is used but current device does not support it'):
+            trainer.fit()
+    else:
+        with pytest.raises(ImportError, match='AMP_FP8 precision is used but TransformerEngine is not installed'):
+            trainer.fit()
```

Comment:
This assumes we're on A100? Is there a way to check device from torch to verify this for first case? 

Idt merge blocking, but would be nice

Reply:
Will add. We have the same check in the precision itself.